From 16bc519d14d48cf59949830d21f38f9ddcff31a3 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 15 Aug 2021 20:51:43 -0400
Subject: [PATCH 4/4] Avoid uint16 warnings by using uint16_t

---
 ColourBrightness.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ColourBrightness.c b/ColourBrightness.c
index 2b82357..f6fb660 100644
--- a/ColourBrightness.c
+++ b/ColourBrightness.c
@@ -60,8 +60,8 @@
 #define ntohs(A) (A)
 #define ntohl(A) (A)
 #elif defined(LITTLE_ENDIAN) && !defined(BIG_ENDIAN)
-#define htons(A) (((uint16)(A) & 0xff00) >> 8 | ((uint16)(A) & 0x00ff) << 8 )
-#define htonl(A) ((((uint32)(A) & 0xff000000) >> 24) | \
+#define htons(A) (((uint16_t)(A) & 0xff00) >> 8 | ((uint16_t)(A) & 0x00ff) << 8 )
+#define htonl(A) ((((uint32_t)(A) & 0xff000000) >> 24) | \
 ((uint32)(A) & 0x00ff0000) >> 8 | \
 ((uint32)(A) & 0x0000ff00) << 8 | \
 ((uint32)(A) & 0x000000ff) << 24)
@@ -1220,8 +1220,8 @@ histograms_struct *ReadHistograms (fullPath *fullPathImages, int numberImages)
   int otherImage;
   int currentImage;
   TIFF **ptrTIFFs = NULL;
-  uint16 samplesPerPixel;
-  uint16 bitsPerSample;
+  uint16_t samplesPerPixel;
+  uint16_t bitsPerSample;
   uint32_t imageLength;
   uint32_t imageWidth;
   char  tempString[512];
-- 
2.30.2

