From b4644df2073a796a15c4249ff8f8a2b42778fa0a Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 15 Aug 2021 21:19:15 -0400
Subject: [PATCH 5/5] Deifne c2pstr for mac aarch64 as well

---
 filter.h  | 3 +++
 sys_mac.h | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/filter.h b/filter.h
index 886e03e..46671ae 100644
--- a/filter.h
+++ b/filter.h
@@ -1090,6 +1090,9 @@ extern sPrefs			*gsPrPtr;
     #ifdef MAC_OS_X_VERSION_10_4
         // MacOSX 10.4 has those functions predefined in Carbon API.
         #include <Carbon/Carbon.h> // CoreServices/TextUtils.h
+    #elif (defined(__arm64__) && defined(__APPLE__)) || defined(__aarch64__)
+        #define     p2cstr( x )
+        #define		c2pstr( x )
     #else
         #define		p2cstr( x )	
         #define		c2pstr( x )
diff --git a/sys_mac.h b/sys_mac.h
index 0c0dddb..b2974fa 100644
--- a/sys_mac.h
+++ b/sys_mac.h
@@ -53,6 +53,9 @@
 
 #ifndef MAC_OS_X_VERSION_10_4
 #define c2pstr(x)  MyCtoPStr(x) // added by Kekus Digital
+#elif (defined(__arm64__) && defined(__APPLE__)) || defined(__aarch64__)
+// Needed on aarch since c2pstr doesn't exist
+#define c2pstr(x)  MyCtoPStr(x) // added by Kekus Digital
 #endif
 
 
-- 
2.30.2

